core/controllers/reader.py

```diff
@@ -1040,3 +1040,34 @@ def get(self):
             'question_dicts': question_dicts
         })
         self.render_json(self.values)
+
+
+class LearnerAnswerDetailsSubmissionHandler(base.BaseHandler):
+    """Handles the learner answer details submission."""
+
+    @acl_decorators.can_submit_answer_details
+    def put(self, entity_id, entity_type):
+        """"Handles the PUT requests. Stores the answer details submitted
+        by the learner.
+        """
+        if not constants.ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE:
+            raise self.PageNotFoundException
+
+        interaction_id = self.payload.get('interaction_id')
+        if entity_type == feconf.ENTITY_TYPE_EXPLORATION:
+            state_name = self.payload.get('state_name')
+            state_reference = (
+                stats_services.get_state_reference_for_exploration(
+                    entity_id, state_name))
+            if not stats_services.validate_interaction_id_with_state_for_exploration( #pylint:disable=line-too-long
```

Comment:
This should really be exp_services.get_interaction_id_for_state(exp_id, state_name) -- and then you can compare it with what's passed in. It doesn't have anything to do with stats.

Reply:
Done